examples/tour.jl

```diff
@@ -0,0 +1,192 @@
+# have been assigned, we want to be able to prescribe and load batches of these data for
+# training and evaluating a machine learning learning model.
+#
+# The RandomBatches type specifies how to iteratively sample from a collection of labeled
+# signals to generate these batches in pseudo-random, iterable fashion.
+#
+# Specifically, RandomBatches constructs batches via the following mechanism:
+# - randomly sample over the signals
+# - for a given signal, randomly select a label
+# - for a given label, randomly select a segment of the signal with that label
+# - for a given segment, randomly select the number of required channels
+#
+# Optionally, we can also weight the sampling of the signals and labels.
+#
+# In this example we are specifying:
+# - 1 channel per sample
+# - 3 samples per batch
+# - Each sample taken over a 60 second window
+# - All signals and labels uniformly weighted
+batches = RandomBatches(labeled_signals, nothing, nothing, 1, 3, Second(60))
+
+# We can now draw down batches of samples from the RandomBatches
+# Calling iterate_batch returns a batch item and the new state of the RNG after iterating once.
+item, new_state = iterate_batch_item(batches, RNG)
+@test item isa BatchItemV2
+@test new_state isa AbstractRNG
+
+# Here, we return and load a single batch item
+# Note that the labels are not necessarily sampled at the same resolution as the signal.
+# This is because the labels are sampled at 0.033 Hz (1 every 30 seconds) while the signal
+# is sampled at 128 Hz.
+x, y = materialize_batch_item(item)
+@test size(x) == (1, 7680)  # 1 channel, 60 seconds @ 128Hz resolution
+@test size(y) == (1, 2)    # 2 labels, one for each 30 second segment
+
+# Additionally, we can draw down the entire batch at once
+batch, new_state = iterate_batch(batches, RNG)
+X, Y = materialize_batch(batch)
+@test size(X) == (1, 7680, 3)  # we now draw 3 samples and concatenate along the 3rd dimension
+@test size(Y) == (1, 2, 3)
+
+# Note that we can continue to draw as many batches as we like by repeatedly passing the
+# new_state back to iterate_batch, much like Julia's Iterator interface
+# https://docs.julialang.org/en/v1/manual/interfaces/#man-interface-iteration
+# In this way, we can dynamically allocate and load batches of data depending on the needs
+# of the model and infrastructure resources that are available.
+batch, new_state = iterate_batch(batches, RNG)
+X2, Y2 = materialize_batch(batch)
+@test X2 != X
+@test Y2 != Y
+
+###
+### Batcher
+###
+
+# Now that we have the mechanism for generating pseudo-random batches, we need to have these
+# dynamically allocated and loaded in parallel across julia processes. This will enable us
+# to scale our training process across a cluster.
+
+# Let's start by adding a few processes and loading the necessary packages
+addprocs(4)
+@everywhere begin
+    using Pkg
+    using OndaBatches
+end
+
+# Note that the "Batch Manager" cannot be assigned to the primary "Manager" node (1) because XXX
```

Comment:
I kept getting errors if I used the primary node...not sure why

Reply:
https://github.com/beacon-biosignals/OndaBatches.jl/issues/25

Reply:
dug into it a bit more and isolated the issue. TLDR there's a bug where the `init_state` gets mutated by `start!` if the batch manager is the root process